### PR TITLE
MDS-163: Enable spacial capabilities (PostGIS)

### PR DESCRIPTION
**Pull request by ActionAnalytics**

### CODE REVIEW

File: migrations/etl/1_Mine_profile_detail_location.sql [hunk 1]
```diff
@@ -149,8 +149,7 @@ BEGIN
     INSERT INTO mine_location(
         mine_location_guid  ,
         mine_guid           ,
-        latitude            ,
-        longitude           ,
+        geom                ,
```
Comment (garywong-bc) [inline migrations/etl/1_Mine_profile_detail_location.sql]: I was assuming an 'expand and then contract'.. so we'd keep mine_location.latitude/longitude until we meed to contract.  This needs to be timed with the other MDS methods that currently work with textual lat/long (e.g. the map rendering of points); if that map still needs textual fields, we can get rid of the two columns but use the PostGIS functions (e.g. `ST_X(geom) as longitude`, `ST_Y(geom) as latitude`)
```

Comment (ActionAnalytics) [reply migrations/etl/1_Mine_profile_detail_location.sql]: Good catch.  Done

File: npm-debug.log [hunk 2]
```diff
@@ -0,0 +1,23 @@
+0 info it worked if it ends with ok
```
Comment (FreshGyan) [inline npm-debug.log]: Can you add a .gitignore entry for this?

File: migrations/Dockerfile.dev [hunk 3]
```diff
@@ -1,4 +1,70 @@
 FROM postgres:9.6-alpine
+MAINTAINER Régis Belson <me@regisbelson.fr>
```
Comment (FreshGyan) [inline migrations/Dockerfile.dev]: Can we update the Maintainer to some email from MDS team?

File: npm-debug.log (hunk 2 above)
Comment (ActionAnalytics) [reply npm-debug.log]: Done

File: migrations/Dockerfile.dev (hunk 3 above)
Comment (ActionAnalytics) [reply migrations/Dockerfile.dev]: Thanks, as per our discussion, removed all Maintainer tags. 

FYI, Maintainer tags are deprecated now, replaced by Author tags.

File: migrations/sql/V1.21__Alter_Mine_Location_Add_Geoms.sql [hunk 4]
```diff
@@ -0,0 +1,5 @@
+-- SRID=3005 is ‘BC Albers’ projection, chosen as the standard projection for this application.
+ALTER TABLE mine_location ADD COLUMN geom GEOMETRY(POINT, 3005);
+UPDATE mine_location SET geom = ST_SetSRID(ST_MakePoint(longitude, latitude),3005) WHERE longitude IS NOT NULL AND latitude IS NOT NULL;
```
Comment (FreshGyan) [inline migrations/sql/V1.21__Alter_Mine_Location_Add_Geoms.sql]: Where is the number 3005 coming from?
If i'm not mistaken all our maps use `4326`. Although im not an expert in these 😅

File: migrations/sql/V1.21__Alter_Mine_Location_Add_Geoms.sql [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+CREATE INDEX mine_location_geom_idx ON mine_location USING GIST(geom);
+--VACUUM ANALYZE mine_location;
```
Comment (FreshGyan) [inline migrations/sql/V1.21__Alter_Mine_Location_Add_Geoms.sql]: Unused comment?

Comment (ActionAnalytics) [reply migrations/sql/V1.21__Alter_Mine_Location_Add_Geoms.sql]: Removed the VACUUM comment.  The 3005 one ads value so leaving it.

File: migrations/sql/V1.21__Alter_Mine_Location_Add_Geoms.sql (hunk 4 above)
Comment (ActionAnalytics) [reply migrations/sql/V1.21__Alter_Mine_Location_Add_Geoms.sql]: We will move forward and create another ticket to discuss the projection we use.

**Pull request merged**